lemur/plugins/base/v1.py

```diff
@@ -109,11 +111,45 @@ def get_resource_links(self):
         """
         return self.resource_links
 
-    @staticmethod
-    def get_option(name, options):
+    def get_option(self, name, options):
+        user_opt = self.get_user_option(name, options)
+        value = user_opt.get("value", user_opt.get("default"))
+        if value is None:
+            return None
+        return self.validate_option_value(name, value)
+
+    def validate_option_value(self, option_name, value):
+        class_opt = self.get_server_options(option_name)
+        if not class_opt:
+            current_app.logger.warning("Plugin option server-defaults not found")
+            return None
+        opt_type = class_opt["type"]
+        if opt_type == "str":
+            validation = class_opt.get("validation")
+            if not validation:
+                return value
+            if not re.match(validation, value):
+                raise ValueError(f"Option '{option_name}' cannot be validated")
+        elif opt_type == "select":
+            available = class_opt.get("available")
+            if available is None:
+                return value
+            if value not in available:
+                raise ValueError(f"Option '{option_name}' doesn't match available options")
+        return value
+
+    def get_server_options(self, name):
+        class_options = getattr(self, "options", [])
```

Comment:
Hm, I noticed I'm [failing Travis](https://travis-ci.com/github/Netflix/lemur/jobs/504920621) due to removing the **@staticmethod** decorator from `get_option` (looks like the AWS plugin uses it statically, most others call with their instance).

The only use of `self` we really need is on this line, to retrieve the server-defined options. Is there a better way to grab those options for validation? If not, maybe we could change the static calls to `get_option` to the util version of it (which appears to do the same thing): https://github.com/Netflix/lemur/blob/master/lemur/plugins/utils.py#L13

Reply:
(trying the latter in https://github.com/Netflix/lemur/pull/3570/commits/d70fac722e6485e66591a37f27c96cec0153f70b)